js/compileShell.py

```diff
@@ -167,7 +167,9 @@ def ensureCacheDir():
     # Expand long Windows paths (overcome legacy MS-DOS 8.3 stuff)
     # This has to occur after the shell-cache directory is created
     if sps.isWin:  # adapted from http://stackoverflow.com/a/3931799
-        winTmpDir = unicode(cacheDir)
+        # Python 3 no longer has the unicode function, see http://python-future.org/compatible_idioms.html#unicode
+        from builtins import str as utext
```

Comment:
`builtins` isn't stdlib in python2. Are you adding a dependency?
What I did in avalanche is this:

    from __future__ import unicode_literals
    if sys.version_info.major == 2:
        # pylint: disable=redefined-builtin,invalid-name
        str = unicode

This changes all old strings to unicode, and if you need raw strings use `b""` or `bytes()` instead (which work in both py2.7 & 3).